File: fabric-audit-agent/adapters/collector.mock.js
import { readFile } from 'node:fs/promises';

/**
 * Mock CollectorPort: reads facts from a fixture file. The real collector
 * (transfer time) calls Fabric/Power BI/Azure APIs and emits the same fact shape.
 * @param {string} fixturePath
 * @returns {{collect: () => Promise<object>}}
 */
export function createMockCollector(fixturePath) {
  return {
    async collect() {
      return JSON.parse(await readFile(fixturePath, 'utf-8'));
    },
  };
}

File: fabric-audit-agent-app/fabric_audit_agent/adapters/collector_mock.py
"""Mock CollectorPort: reads facts from a fixture JSON file. Port of ``adapters/collector.mock.js``.

The real collector (``collector_rest``) calls Fabric/Power BI/Azure APIs and emits the same
fact shape; this one just loads a fixture so the pipeline runs fully offline.
"""
import json


def create_mock_collector(fixture_path):
    def collect():
        with open(fixture_path, "r", encoding="utf-8") as fh:
            return json.load(fh)

    return {"collect": collect}
